airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/SnowflakeDestination.kt

```diff
@@ -317,8 +317,9 @@ constructor(
 
 fun main(args: Array<String>) {
     IntegrationRunner.addOrphanedThreadFilter { t: Thread ->
-        if (IntegrationRunner.getThreadCreationInfo(t) != null) {
-            for (stackTraceElement in IntegrationRunner.getThreadCreationInfo(t)!!.stack) {
+        val threadCreationInfo = IntegrationRunner.getThreadCreationInfo(t)
```

Comment:
there was a race condition here, where getThreadCreationInfo would return nun-null on the 1st call but null on the 2nd call.

Reply:
I know why :)
The reason being that between the time the IntegrationRunner looks at all live thread and the time we hit this code, the thread could have completed, and all its threadLocal could have been garbage collected, because their only durable instance is held in the thread... so what happens is that we called `getThreadCreationInfo(t)`, which was not null. Then the thread completed, the map was cleared, and we call again `getThreadCreationInfo(t)`, but at this point it returns null... The magic of parallel computing

Reply:
thought about it a bit more, and I think the proper fix is not to pass in a `Thread` to the filter, but instead a `ThreadInfo`, which would contain a `Thread`, but also all the info we might need (like `currentStackTrace` and `ThreadCreationInfo`), so that the thread itself can die after this is getting called without causing any side effects.